### ISSUE TITLE
Support the latest `vsphere-automation-sdk-python` as required by `community.vmware`

### ISSUE THREAD

**Issue by kurokobo** [ansible/awx-ee#137]
Just to inform, daily builds seem to be failing since Sep. 29.

I have not looked into it in detail, but it seems to be taking an extremely long time to resolve dependencies by pip and is facing a 6 hour timeout.

https://github.com/ansible/awx-ee/actions

**EDIT AFTER CASE CLOSED:**
The upstream `ansible-runner` image still needs to be fixed, but at this time as @shanemcd did in #141 adding `ansible-runner>=2.2.1` to `requirements.txt` for Ansible Builder is the simplest workaround.

**Comment by kurokobo** [comment]
This seems to be caused by this commit on `vsphere-automation-sdk-python`: https://github.com/vmware/vsphere-automation-sdk-python/commit/4b49fbbcf3169020966b9c4c44d8a5454cca7bec
This commit updated `whl`s in the repository, and updated `whl`s have updated dependencies.

My guess is:

- `awx-ee` tries to install `community.vmware`
- `community.vmware` has `requirements.txt` that refer upstream `https://github.com/vmware/vsphere-automation-sdk-python`
- Ansible Builder merges all `requirements.txt` from all Collections and user-defined one
- Pip tries to install `https://github.com/vmware/vsphere-automation-sdk-python` and `whl`s in the repository that updated few days ago by above commit
- Dependencies for Pip modules had been changed and can't be resolved anymore, therefore build has been failed

Of course the latest `vsphere-automation-sdk-python` should be supported, but as a temporary (and super dirty) workaround, we can _pin_ the version of `vsphere-automation-sdk-python` by following patch. 

<details>
<summary>
<i>EDIT: Outdated workaround</i>
</summary>


```patch
diff --git a/tox.ini b/tox.ini
index e773891..28a9d44 100644
--- a/tox.ini
+++ b/tox.ini
@@ -10,21 +10,29 @@ deps = -r{toxinidir}/requirements.txt
 [testenv:podman]
 passenv =
   HOME
-whitelist_external =
+whitelist_externals =
   true
+  sed
+  podman
 commands =
   /bin/bash -c "podman rmi quay.io/ansible/awx-ee:latest || true"
-  ansible-builder build -v3 -c . -t quay.io/ansible/awx-ee {posargs}
+  ansible-builder create -v3 -c .
+  sed -ri 's/^(RUN ansible-builder introspect.*)$/\1; sed -i \'s\/vsphere-automation-sdk-python.git\/vsphere-automation-sdk-python.git@46aa57de24420ead6af13fe4c8783f1ff66ffcfd\/g\' \/tmp\/src\/requirements.txt/g' Containerfile
+  podman build {posargs} .
 
 
 [testenv:docker]
 passenv =
   HOME DOCKER_BUILDKIT
-whitelist_external =
+whitelist_externals =
   true
+  sed
+  docker
 commands =
   /bin/bash -c "podman rmi quay.io/ansible/awx-ee:latest || true"
-  ansible-builder build -v3 -c . -t quay.io/ansible/awx-ee {posargs} --container-runtime=docker
+  ansible-builder create -v3 -c .
+  sed -ri 's/^(RUN ansible-builder introspect.*)$/\1; sed -i \'s\/vsphere-automation-sdk-python.git\/vsphere-automation-sdk-python.git@46aa57de24420ead6af13fe4c8783f1ff66ffcfd\/g\' \/tmp\/src\/requirements.txt/g' Containerfile
+  docker build {posargs} .
 
 [testenv:check-diff]
 passenv =
```
</details>

**Comment by thedatabaseme** [comment]
@kurokobo Removing community-vmware fixes the issue for me. So within there, the problem seems to be.

**Comment by kurokobo** [comment]
Instead of using temporary workaround, to support updated `vsphere-automation-sdk-python`, there seems to be additional considerations:

- Updated `vsphere-automation-sdk-python` requires `pyopenssl==22.0.0`, but [it is not compatible with `ansible-runner` (that requires `pyopenssl<20.0.0`)](https://github.com/ansible/ansible-runner/issues/1138). This should be fixed in ansible-runner.
- Updated `vsphere-automation-sdk-python` requires `cryptography==36.0.0`, but [`awx-ee` requires `cryptography==37.0.4`](https://github.com/ansible/awx-ee/blob/f3a6157237093f01a7b7c3b92cc329cd23cc5be2/_build/requirements.txt#L2). This should be fixed in this `awx-ee` repository.

**Comment by mumohamm** [comment]
Can you please let me know how to apply the patch?

**Comment by kurokobo** [comment]
@mumohamm 
<details>
<summary>
<i>EDIT: Outdated workaround</i>
</summary>
You can divide following `ansible-builder build` command;

```bash
ansible-builder build --tag registry.example.com/ansible/ee:2.12-custom --container-runtime docker --verbosity 3
```

into following three steps;

```bash
# Create context files including Dockerfile.
# This will create `context` directory on current directory.
ansible-builder create --verbosity 3

# Modify Dockerfile to edit `requirements.txt` used during build process
# This will append `; sed ... /tmp/src/requirements.txt` command to modify `requirements.txt` at the end of existing `ansible-builder introspect` command
sed -ri 's/^(RUN ansible-builder introspect.*)$/\1; sed -i s\/vsphere-automation-sdk-python.git\/vsphere-automation-sdk-python.git@46aa57de24420ead6af13fe4c8783f1ff66ffcfd\/g \/tmp\/src\/requirements.txt/g' context/Dockerfile

# Build EE
docker build -t registry.example.com/ansible/ee:2.12-custom context
```
</details>

**Comment by thedatabaseme** [comment]
@kurokobo FYI, `cryptography` was fixed to this Version due to issue https://github.com/ansible/awx-ee/issues/134

**Comment by djyasin** [comment]
@shanemcd will be happy to help whoever picks this issue up.

**Comment by relrod** [comment]
Fixed by #141

**Comment by kurokobo** [comment]
@shanemcd @relrod 
Thanks for working on this :)
